### PR TITLE
Fixes for ts4

### PR DESCRIPTION
**Pull request by josh-cain**
When importing this library into a Typescript 4 project, `tsc` will fail.  This PR is an attempt at fixes to remedy.  Some fixes:

- `wrapCallbackApi` was calling `Promise.resolve` with an array.  Two problems: 1) Only the first argument will ever be passed by the resolved promise, and 2) when stricter typing is enforced, passing an array causes `tsc` to fail.
- `Timeout`'s `CallablePromise` did not properly enumerate the callback parameter typing

Also fixed a gratuitous use of `args` as a variable name:

Screen Shot 2021-02-02 at 4 18 08 PM

### CODE REVIEW

File: src/Disyuntor.ts [hunk 1]
```diff
@@ -156,15 +155,19 @@ export function wrapCallbackApi<T extends (...args: any[]) => void>(
...
       });
     }).then(
-      (...args) => {
-        callback(null, ...args)
+      (args) => {
+        if (Array.isArray(args)) {
+          callback(null, ...args);
+        } else {
+          callback(null, args);
+        }
```
Comment (josh-cain) [inline src/Disyuntor.ts:171]: I don't particularly love this if/else.  Anyone have some syntactic sugar that might do this for me?

Comment (ktraff-auth0) [reply src/Disyuntor.ts:171]: not sure this is any any better 🤷 

```javascript
callback(null, ...(Array.isArray(args) ? args : [args]))
```

File: src/Options.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export namespace Options {
 
     /**
      * The maximum time the circuit remains open before doing a new attempt.
-     * Defaults to 60s
+     * Defaults to 30s
```
Comment (josh-cain) [inline src/Options.ts:29]: The effective default was 30s per the code, so I used the actual value and updated documentation.

File: src/Disyuntor.ts (hunk 1 above)
Comment (jfromaniello) [reply src/Disyuntor.ts:171]: I don't know about the syntax sugar but what we don't want is to unexpectedly spread the arguments .. Let's say that the function we are "protecting" does something like this
```js
function test(callback) {
 setImmediate( () => callback(null, [1,2,3] ) );
}
```
we want to preserve the array. Excuse me is this is obvious but with so many `...` is hard to follow 🤣

Comment (josh-cain) [reply src/Disyuntor.ts:171]: Good call out @jfromaniello !  I had *not* considered this case explicitly.

However, I just wrote a test using something like the case you gave here... and I believe it to be working.  Here's why: when we hit the Array.isArray(args) check, we actually have a single-element array (containing the multi-value array we passed to the callback) like this:

Screen Shot 2021-02-04 at 3 32 09 PM

At this point I'm not certain that we'll ever need the `else` block though 🤔 

EDIT: pushed test code as well for examination.

**Review by jfromaniello — APPROVED** [review]
Looks good Josh!
